PWGJE/Tasks/jetSpectraESE.cxx

```diff
@@ -0,0 +1,147 @@
+
+  Configurable<std::string> eventSelections{"eventSelections", "sel8", "choose event selection"};
+  Configurable<std::string> trackSelections{"trackSelections", "globalTracks", "set track selections"};
+
+  int eventSelection = -1;
+  int trackSelection = -1;
+
+  void init(o2::framework::InitContext&)
+  {
+    eventSelection = jetderiveddatautilities::initialiseEventSelection(static_cast<std::string>(eventSelections));
+    trackSelection = jetderiveddatautilities::initialiseTrackSelection(static_cast<std::string>(trackSelections));
+  }
+
+  Filter jetCuts = aod::jet::pt > jetPtMin&& aod::jet::r == nround(jetR.node() * 100.0f);
+
+  int spCode(const int qPerc)
+  {
+    if (qPerc < 0)
+      return -1;
+
+    if (qPerc <= qLimits->at(0)) {
+      return 0;
+    } else if (qPerc >= qLimits->at(1)) {
+      return 1;
+    } else {
+      return -1;
+    }
+  }
+  std::string jetplane(float deltaPhi)
+  {
+    std::string s = "";
+    if (deltaPhi < TMath::Pi() / 6.) {
+      s = "in";
+    } else if (deltaPhi > TMath::Pi() / 3.) {
+      s = "out";
+    }
+
+    return s;
+  }
+
+  // void processESEDataCharged(JColwESE::iterator const& collision, soa::Filtered<aod::ChargedJets> const& jets, aod::BCsWithTimestamps const&)
+  void processESEDataCharged(soa::Join<JetCollisions, aod::JCollisionPIs>::iterator const& collision,
+                             soa::Join<aod::Collisions, aod::CentFT0Cs, aod::qVecFT0Cs, aod::qPercentileFT0Cs> const&,
+                             soa::Filtered<aod::ChargedJets> const& jets)
+  {
+    auto originalCollision = collision.collision_as<soa::Join<aod::Collisions, aod::CentFT0Cs, aod::qVecFT0Cs, aod::qPercentileFT0Cs>>();
+
+    float vPsi2 = FFitWeights::EventPlane(originalCollision, 2);
+
+    auto qPerc = originalCollision.qPERCFT0C();
+    if (qPerc[0] > 0)
+      registry.fill(HIST("h_Cent_q2Percs"), originalCollision.centFT0C(), qPerc[0]);
+
+    int code = spCode(qPerc[0]);
+
+    registry.fill(HIST("h_collisions"), 0.5);
+    if (!jetderiveddatautilities::selectCollision(collision, eventSelection)) {
+      return;
+    }
+
+    registry.fill(HIST("h_collisions"), 1.5);
+    float plane{0};
+    std::string pl{""};
+    for (auto const& jet : jets) {
+
+      plane = vPsi2 - jet.phi();
+      pl = jetplane(plane);
+      registry.fill(HIST("h_jet_pt"), jet.pt());
+
+      if (code < 0)
+        continue;
+
+      if (code == 0)
+        registry.fill(HIST("h_jet_pt_q2_0"), jet.pt());
+      if (code == 1)
+        registry.fill(HIST("h_jet_pt_q2_1"), jet.pt());
+      if (pl == "in")
+        registry.fill(HIST("h_jet_pt_in"), jet.pt());
+      if (pl == "out")
+        registry.fill(HIST("h_jet_pt_out"), jet.pt());
+    }
+  }
+  PROCESS_SWITCH(JetSpectraEseTask, processESEDataCharged, "process self contained collisions", true);
+};
+
+WorkflowSpec defineDataProcessing(ConfigContext const& cfgc) { return WorkflowSpec{adaptAnalysisTask<JetSpectraEseTask>(cfgc, TaskName{"jet-spectra-ese"})}; }
```

Comment:
Please don't use `TaskName` if not needed by templatation. If you just name your struct `JetSpectraEse`, you will get the same result.

Reply:
Thanks